frontend/src/components/auth/ResetPassword.tsx

```diff
@@ -14,17 +22,23 @@ const ResetPassword = ({ email }: ResetPasswordProps) => {
     return re.test(String(emailString).toLowerCase());
   };
 
+  type ResetPassword = { ok: boolean };
+  const [resetPassword] = useMutation<{ resetPassword: ResetPassword }>(
```

Comment:
> Am I using Graphene incorrectly -> can we do resetUser(email: String!): Boolean! instead of resetUser(email: String!): ResetUser?

For context: Emma tried `useMutation<{ ok: boolean }>` but ran into this problem:
```
let res = await resetPassword({ variables: { email: resetEmail } });
console.log(res);
// {data: resetPassword: {ok: true, __typename: "ResetPassword"}}
console.log(res.data.resetPassword);
// Property 'resetPassword' does not exist on type '{ ok: boolean; }'
```
The conclusion is that `useMutation` sets what Typescript expects res to look like. While Emma's code works, the pattern feels odd. `Logout.tsx` in the graphql starter-code doesnt have to follow something like this.

@lynn-kim @kevinwang2040 What have y'all been experiencing? I don't have a good answer for why this is going on.

Reply:
The pattern in the frontend is a result of how the GraphQL mutation responds, ie. `resetUser(email: String!): ResetUser` instead of `resetUser(email: String!): Boolean!`? -- so I think it's a limitation of how Graphene has mutations and I haven't found a workaround

Related Q: https://github.com/graphql-python/graphene/issues/497

Reply:
As an experiment, you could try copying your `ResetPassword.tsx` (and other necessary files/changes) to the graphql configured starter-code and seeing what works. Im curious if the typescript graphql server from starter-code also follows the `data:resetuser:ok` pattern.

Reply:
Their frontend looks like:
```
const RESET_PASSWORD = gql`
  mutation ResetPassword($email: String!) {
    resetPassword(email: $email)
  }
`;

const [resetPassword] = useMutation<{ resetPassword: boolean }>(
  RESET_PASSWORD,
);

const onResetPasswordClick = async () => {
  await resetPassword({ variables: { email: authenticatedUser?.email } });
};
```

which matches their backend, where the mutation is `resetPassword(email: String!): Boolean!` and they don't have to filter through a `ResetPassword` object to get the result?

Reply:
rip